services/provider/server/server.go

```diff
@@ -1042,3 +1043,130 @@ func (s *OCSProviderServer) PeerStorageCluster(ctx context.Context, req *pb.Peer
+
+		if owner == nil {
+			continue
+		}
+
+		if owner.UID != types.UID(req.StorageClusterUID) {
+			klog.Errorf("storageCluster specified on the req does not own the Client")
+			response.Error = append(response.Error,
+				&pb.StorageClientInfoError{
+					ClientID: req.ClientIDs[i],
+					Code:     pb.ErrorCode_Internal,
+					Message:  "storageCluster specified on the req does not own the Client",
+				},
+			)
+			continue
+		}
+
+		rnsList := &rookCephv1.CephBlockPoolRadosNamespaceList{}
+		err = s.client.List(
+			ctx,
+			rnsList,
+			client.InNamespace(s.namespace),
+			client.MatchingLabels{controllers.StorageConsumerNameLabel: consumer.Name},
+			client.Limit(2),
+		)
+		if err != nil {
+			klog.Error(err)
+			response.Error = append(response.Error,
+				&pb.StorageClientInfoError{
+					ClientID: req.ClientIDs[i],
+					Code:     pb.ErrorCode_Internal,
+					Message:  "failed to get radosnamespaces",
+				},
+			)
+			continue
+		}
+		if len(rnsList.Items) > 1 {
+			response.Error = append(response.Error,
+				&pb.StorageClientInfoError{
+					ClientID: req.ClientIDs[i],
+					Code:     pb.ErrorCode_Internal,
+					Message:  "invalid number of radosnamespace found for the Client",
+				},
+			)
+		} else if len(rnsList.Items) == 1 {
+			response.ClientsInfo = append(response.ClientsInfo, &pb.ClientInfo{ClientID: req.ClientIDs[i], RadosNamespace: rnsList.Items[0].Name})
+		} else {
+			response.ClientsInfo = append(response.ClientsInfo, &pb.ClientInfo{ClientID: req.ClientIDs[i]})
+		}
+	}
+
+	return response, nil
+}
+
+func (s *OCSProviderServer) GetBlockPoolsInfo(ctx context.Context, req *pb.BlockPoolsInfoRequest) (*pb.BlockPoolsInfoResponse, error) {
+	response := &pb.BlockPoolsInfoResponse{BlockPoolsInfo: []*pb.BlockPoolInfo{}, Error: []*pb.BlockPoolInfoError{}}
+
+	for i := range req.BlockPoolNames {
+		cephBlockPool := &rookCephv1.CephBlockPool{}
+		cephBlockPool.Name = req.BlockPoolNames[i]
+		cephBlockPool.Namespace = s.namespace
+		err := s.client.Get(ctx, client.ObjectKeyFromObject(cephBlockPool), cephBlockPool)
+		if err != nil {
+			if !kerrors.IsNotFound(err) {
+				response.Error = append(response.Error,
+					&pb.BlockPoolInfoError{
+						BlockPoolName: cephBlockPool.Name,
+						Code:          pb.ErrorCode_Internal,
+						Message:       err.Error(),
+					},
+				)
+			}
+			continue
+		}
+
+		var mirroringToken string
+
+		if cephBlockPool.Spec.Mirroring.Enabled &&
+			cephBlockPool.Status.Info != nil &&
+			cephBlockPool.Status.Info[mirroringTokenKey] != "" {
+			secret := &corev1.Secret{}
+			secret.Name = cephBlockPool.Status.Info[mirroringTokenKey]
+			secret.Namespace = s.namespace
+			err := s.client.Get(ctx, client.ObjectKeyFromObject(secret), secret)
+			if err != nil {
+				errMsg := fmt.Sprintf(
+					"failed to get bootstrap secret %s for CephBlockPool %s: %v",
+					cephBlockPool.Status.Info[mirroringTokenKey],
+					cephBlockPool.Name,
+					err,
+				)
+				klog.Error(errMsg)
```

Comment:
Not finding the secret should not be treated as an error from the perspective of the RPC call. It might be an error for a controller but not for the server